src/Vault.sol

```diff
@@ -102,14 +102,18 @@ contract Vault is IVault, VaultToken, Ownable {
 
     /// @inheritdoc IVault
     function take(Currency currency, address to, uint256 amount) external override isLocked {
-        SettlementGuard.accountDelta(msg.sender, currency, -(amount.toInt128()));
-        currency.transfer(to, amount);
+        unchecked {
+            SettlementGuard.accountDelta(msg.sender, currency, -(amount.toInt128()));
+            currency.transfer(to, amount);
+        }
     }
 
     /// @inheritdoc IVault
     function mint(address to, Currency currency, uint256 amount) external override isLocked {
-        SettlementGuard.accountDelta(msg.sender, currency, -(amount.toInt128()));
-        _mint(to, currency, amount);
+        unchecked {
+            SettlementGuard.accountDelta(msg.sender, currency, -(amount.toInt128()));
+            _mint(to, currency, amount);
+        }
```

Comment:
1. amount is positive 
2. if `amount.toInt128` doesn't revert, then amount is a valid int128
3. combine 1 & 2, converting from positive to negative will never go overflow